src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -66,5 +83,196 @@ void ConnectivityManagerImpl::_OnPlatformEvent(const ChipDeviceEvent * event)
+    }
+
+    if (iface != NULL && err == NULL)
+    {
+        mWpaSupplicant.iface = iface;
+        mWpaSupplicant.state = GDBusWpaSupplicant::WPA_INTERFACE_CONNECTED;
+        ChipLogProgress(DeviceLayer, "wpa_supplicant: connected to wpa_supplicant interface proxy: %p", iface);
+    }
+    else
+    {
+        if (err != NULL)
+            ChipLogProgress(DeviceLayer, "wpa_supplicant: failed to create wpa_supplicant1 interface proxy %s: %s",
+                            mWpaSupplicant.interfacePath, err->message);
+        else
+            ChipLogProgress(DeviceLayer, "wpa_supplicant: failed to create wpa_supplicant1 interface proxy %s",
+                            mWpaSupplicant.interfacePath);
+
+        mWpaSupplicant.state = GDBusWpaSupplicant::WPA_NOT_CONNECTED;
+    }
+
+    if (err != NULL)
+        g_error_free(err);
+}
+
+void ConnectivityManagerImpl::_OnWpaInterfaceReady(GObject * source_object, GAsyncResult * res, gpointer user_data)
+{
+    GError * err = NULL;
+
+    gboolean result =
+        wpa_fi_w1_wpa_supplicant1_call_get_interface_finish(mWpaSupplicant.proxy, &mWpaSupplicant.interfacePath, res, &err);
+    if (result)
+    {
+        mWpaSupplicant.state = GDBusWpaSupplicant::WPA_GOT_INTERFACE_PATH;
+        ChipLogProgress(DeviceLayer, "wpa_supplicant: WiFi interface: %s", mWpaSupplicant.interfacePath);
+
+        wpa_fi_w1_wpa_supplicant1_interface_proxy_new_for_bus(G_BUS_TYPE_SYSTEM, G_DBUS_PROXY_FLAGS_NONE, "fi.w1.wpa_supplicant1",
+                                                              mWpaSupplicant.interfacePath, NULL, _OnWpaInterfaceProxyReady, NULL);
+    }
+    else
+    {
+        if (err != NULL)
+            ChipLogProgress(DeviceLayer, "wpa_supplicant: can't find interface %s: %s", CHIP_DEVICE_CONFIG_WIFI_STATION_IF_NAME,
+                            err->message);
+        else
+            ChipLogProgress(DeviceLayer, "wpa_supplicant: can't find interface %s", CHIP_DEVICE_CONFIG_WIFI_STATION_IF_NAME);
+
+        mWpaSupplicant.state = GDBusWpaSupplicant::WPA_NO_INTERFACE_PATH;
+
+        if (mWpaSupplicant.interfacePath)
+        {
+            g_free(mWpaSupplicant.interfacePath);
+            mWpaSupplicant.interfacePath = NULL;
+        }
+    }
+
+    if (err != NULL)
+        g_error_free(err);
+}
+
+void ConnectivityManagerImpl::_OnWpaInterfaceAdded(WpaFiW1Wpa_supplicant1 * proxy, const gchar * path, GVariant * properties,
+                                                   gpointer user_data)
+{
+    if (mWpaSupplicant.interfacePath)
+        return;
+
+    mWpaSupplicant.interfacePath = const_cast<gchar *>(path);
+    if (mWpaSupplicant.interfacePath)
+    {
+        mWpaSupplicant.state = GDBusWpaSupplicant::WPA_GOT_INTERFACE_PATH;
+        ChipLogProgress(DeviceLayer, "wpa_supplicant: WiFi interface added: %s", mWpaSupplicant.interfacePath);
+
+        wpa_fi_w1_wpa_supplicant1_interface_proxy_new_for_bus(G_BUS_TYPE_SYSTEM, G_DBUS_PROXY_FLAGS_NONE, "fi.w1.wpa_supplicant1",
```

Comment:
Can we move the interface string to a constant?

namespace {
const char kWpaSupplicantInterface[] = "fi.w1.wpa_supplicant1";
}

Reply:
Ack